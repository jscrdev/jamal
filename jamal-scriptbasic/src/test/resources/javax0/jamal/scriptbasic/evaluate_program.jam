{@import res:javax0/jamal/scriptbasic/scriptbasic.jim}
{expr 13+1}{@define start=1}{@define end=13}

{basic for i={start} to {end}
if i%2 = 1 then
  oddity = "odd"
else
  oddity = "even"
endif
print i,". is an ",oddity," number\n"
next
}
{@eval/ScriptBasic PRINT 55+13}