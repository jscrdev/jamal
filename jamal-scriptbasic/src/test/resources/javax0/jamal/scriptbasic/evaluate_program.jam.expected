
------------------------------------------------------------------------------------------------------------------------
here we have the source encoding in <properties></properties> with default encoding


<properties>
    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
</properties>
------------------------------------------------------------------------------------------------------------------------

here we have the source encoding in <properties></properties> with given encoding


<properties>
    <project.build.sourceEncoding>ISO-Latin-2</project.build.sourceEncoding>
</properties>

here we start the properties and inside
<properties>
------------------------------------------------------------------------------------------------------------------------
encoding is a single property with default encoding

<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>

------------------------------------------------------------------------------------------------------------------------
encoding is a single property with given encoding

<project.build.sourceEncoding>ISO-Latin-2</project.build.sourceEncoding>
</properties>


------------------------------------------------------------------------------------------------------------------------
here we have the source encoding in <properties></properties> with default encoding
as redefinition was protected by block


<properties>
    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
</properties>

------------------------------------------------------------------------------------------------------------------------
here we have the source encoding in <properties></properties> with given encoding
as redefinition was protected by block


<properties>
    <project.build.sourceEncoding>ISO-Latin-2</project.build.sourceEncoding>
</properties>

------------------------------------------------------------------------------------------------------------------------
here we would have double properties, because define is executed only after the user defined macro but we protect it
with ident

<properties>
<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
<project.build.sourceEncoding>ISO-Latin-2</project.build.sourceEncoding>

</properties>